roles/java/tasks/main.yml

```diff
@@ -22,11 +22,15 @@
           - --xform=s|[^/]*|{{ java_home | basename }}|
 
     - name: Add setenv bash file
-      template:
-        src: setenv.sh
-        dest: "{{ config_folder }}/setenv.sh"
+      blockinfile:
+        path: "{{ config_folder }}/setenv.sh"
+        marker: "# {mark} JAVA ENV VARS"
+        create: yes
+        block: |
+          #!/bin/sh
+          #
```

Comment:
I've the feeling it's more robust if we define a separate task which deploy a static file with just the shebang instead of relying that this will be forever the first task creating that file (`create: yes` will mask the file created before this task)

Reply:
In that case the issue remains. The java role being a dependency of other role using the setenv.sh, the playbook would endup wiping the setenv.sh file for every host with at least 2 roles depending on java.
The alternative I really would like to implement at some point is to completely stop relying on such a hacky setenv.sh script and instead make sure to populate the systemd unit files with the right ENV vars (either in the unit file itself or using an env file).

Reply:
yeah you are right, probably blockinfile should be used only for files deployed externally from ansible and we should instead have a template for that file

Reply:
created OPSEXP-1430 to track that